src/node/build.ts

```diff
@@ -27,8 +27,9 @@ function DefaultIncludedRoutes(paths: string[], _routes: Readonly<RouteRecordRaw
 }
 
 export async function build(ssgOptions: Partial<ViteSSGOptions> = {}, viteConfig: InlineConfig = {}) {
-  const mode = process.env.MODE || process.env.NODE_ENV || ssgOptions.mode || 'production'
-  const config = await resolveConfig(viteConfig, 'build', mode, mode !== 'development' ? 'production' : 'development')
+  const nodeEnv = process.env.NODE_ENV || 'production'
+  const mode = process.env.MODE || ssgOptions.mode || nodeEnv
+  const config = await resolveConfig(viteConfig, 'build', mode, nodeEnv)
```

Comment:
This seems to miss the fix in https://github.com/antfu/vite-ssg/pull/362

Reply:
After reviewing the fix in #362, I noticed that this modification has compromised the flexibility of adjusting NODE_ENV.

For example:

```bash
nr cross-env NODE_ENV=development vite-ssg build --mode ssg
```

I believe that the logic for NODE_ENV and MODE should not be merged together.